ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/gcc_10/tasks/main.yml

```diff
@@ -0,0 +1,42 @@
+---
+##########
+# gcc_10 #
+##########
+
+- name: Check if gcc 10.3 is installed on RHEL/CentOS/Ubuntu16
+  shell: /usr/local/gcc10/bin/gcc-10.3 --version 2>&1 > /dev/null
+  ignore_errors: yes
+  register: gcc10_installed
+  when: ansible_distribution == "RedHat" or ansible_distribution == "CentOS" or ansible_distribution == "openSUSE" or (ansible_architecture == "armv7l" and ((ansible_distribution == "Ubuntu" and ansible_distribution_major_version == "16") or ansible_distribution == "Debian"))
+  tags: gcc_10
+
+# Unable to check the checksum of the binary as it'll be different for each architecture's tar.xz file
```

Comment:
Could we not query these, based on the `ansible_architecture` variable, and store the expected checksum in `ci.adoptopenjdk.net/userContent/gcc` ?

Reply:
We could, but then if the tarball is compromised the checksum from the same place could be too (I'm a little more inclined to trust the ones from our servers as opposed to third party ones. We could probably use variables to store the expected checksums directly in the role itself.

However, that's probably a separate PR if we go for that, since what I've done here for 10.3 is the same as what we're doing for the earlier GCC versions and I don't want to hold this up because of it.

Reply:
Raised at https://github.com/adoptium/infrastructure/issues/2423

Reply:
Good point on storing them in the same place, happy to put this off for now, then :+1: